### PR TITLE
Various macOS and win32 fixes

### LINKED ISSUE #10124: Disable window title buttons
**Is your feature request related to a problem? Please describe.**

The CanResize value does not turn off the Linux maximize button. And for a more effective solution, I think there should be support for disabling maximize, minimize and close buttons.

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

This one was quite a journey. I originally intended to fix #10124 on macOS but in doing so ended up shaving many yaks. This PR is probably better reviewed as a set of commits; I considered splitting it up into many PRs but not convinced that wouldn't just increase review workload so decided to keep it like this. Let me know if it would be best split up.

- After adding extra integration tests in 571b7746eac6cc694aa586ec83d48ad5c9f4b9ac and 2951b80c39601a8368cc902f76ef4f01bc9307c5 I noticed that #10153 had caused a problem on macOS whereby when a window has `CanResize == false` then showing the window causes it to be shown on the bottom left of the screen instead of the normal/requested position. Was caused because it seems that `[NSWindow isZoomed]` always returns `true` when the window has no borders unless you implement a method with a boilerplate implementation: e7a6d6fbc690b96af757199c98ff1f324f3d75ec
- To prevent regressions of this behavior, added tests for `CanResize == false` to the integration tests
- Which revealed another problem: programmatically making a non-resizable window fullscreen on macOS didn't work because 
`[NSWindow setIsZoomed]` requires that the window is resizable by the user in order to work. Fixed in 38aaadf92db4e4957a3ccab2365937c215e3c89a
- While investigating this, noticed that `WindowBaseImpl::CreateNSWindow` called the virtual method `WindowBaseImpl::GetStyle` from a constructor, meaning that the overridden implementation wasn't called. Fix this in 5346344d12fa314c7317f90f03c8e798ee4ef6cc and also removed some unneeded code (this method is only called once, from the ctor, no need to check for an existing `Window`)
- Given that I'm touching this code, I've always found the naming of `WindowBaseImpl::GetStyle` confusing: renamed it to `CalculateStyleMask` and made it abstract to prevent the previous problem: d40041f02dd8ae5e53074e3b4c3e47b420ff9457
- Now I can finally fix #10124 on macOS in b609ba58a8e445ff68ddbcd64d9ab3d9d4f6cb6c and 6a1910172c2a060f7ef67d875af03064db768d5f
- Now on macOS we still have two `WindowState` integration tests failing and a couple of flaky tests which are now at least failing more consistently
- Make the two `WindowState` tests pass by adding a hack for more strange macOS behavior in a705f546bba1f3fb973aae2cdeb9ed8d08cf4f16 and 1abb1abaf5f03ee87f66946acc3b2fc709a45d73
- Fix the flaky integration tests by tweaking our window z-ordering code in 784c380c604b6fd6a139abd7233aaf668661bfd0
- Now we have some integration tests failing on win32: when programmatically maximizing a window with `CanResize == false`, the window was sized incorrectly (didn't quite fill the screen). Added a hackfix for this strange win32 behavior in 339b1e92bdc95c8b07b6c7f410744719883ffafe

## Fixed issues

Fixes (on macOS) #10124

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0029741-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by Sorien** [comment]
maybe i remember it wrong but in windows you should listen for https://learn.microsoft.com/en-us/windows/win32/winmsg/wm-getminmaxinfo and return proper size

**Comment by grokys** [comment]
@Sorien interesting. I'm not expert on the win32 API so have no idea if the way we're currently doing it is better or worse. Might be worth experimenting changing it, but I think this PR has enough stuff in it for now. If you'd like to open a PR which fixes it, please do!

**Comment by Sorien** [comment]
@grokys I'm not either, but maybe will find some free time to play with it, I'd say that your solution can have visual glitches, but is better than nothing, as far as I understand it right it could be the same as https://stackoverflow.com/questions/42019146/wpf-window-without-border-expands-outside-of-screen-in-maximize so WPF has the same problem.

File: native/Avalonia.Native/src/OSX/WindowImpl.mm [hunk 1]
```diff
@@ -612,3 +597,25 @@
...
+    NSButton* closeButton = [Window standardWindowButton:NSWindowCloseButton];
+    NSButton* miniaturizeButton = [Window standardWindowButton:NSWindowMiniaturizeButton];
+    NSButton* zoomButton = [Window standardWindowButton:NSWindowZoomButton];
+
+    [closeButton setHidden:!hasTrafficLights];
+    [closeButton setEnabled:_isEnabled];
+    [miniaturizeButton setHidden:!hasTrafficLights];
+    [miniaturizeButton setEnabled:_isEnabled];
+    [zoomButton setHidden:!hasTrafficLights];
+    [zoomButton setEnabled:_isEnabled && _canResize];
```
Comment (maxkatz6) [inline native/Avalonia.Native/src/OSX/WindowImpl.mm:620]: Do we need manually change these buttons visibility after we set correct styleMask?

**Review by maxkatz6 — APPROVED** [review]
lgtm in overall

File: native/Avalonia.Native/src/OSX/WindowImpl.mm (hunk 1 above)
Comment (grokys) [reply native/Avalonia.Native/src/OSX/WindowImpl.mm:620]: Yeah the button visibility and `styleMask` are two separate things.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0030025-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**